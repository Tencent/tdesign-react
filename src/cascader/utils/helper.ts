import { TreeNode, CascaderContextType, TreeNodeValue, TreeOptionData } from '../interface';

/**
 * 获取node的全部路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}

/**
 * treeValue计算方法
 * @param value
 * @returns
 */
export const getTreeValue = (value: CascaderContextType['value']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && typeof value[0] === 'object') {
      treeValue = (value as TreeOptionData[]).map((val) => val.value);
    } else if (value.length) {
      treeValue = value as TreeNodeValue[];
    }
  } else if (value) {
    if (typeof value === 'object') {
      treeValue = [(value as TreeOptionData).value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};

/**
 * 判断node是否在页面上
 * @param node Node
 */
export function isInPage(node: Node) {
  return node === document.body ? false : document.body.contains(node);
}

export default {
  getFullPathLabel,
};
