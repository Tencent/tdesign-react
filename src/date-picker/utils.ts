import chunk from 'lodash/chunk';
import { DateValue } from './type';
import { Styles } from '../common';

/**
 * next Month
 * @since version V1.0.0
 * @param { Date } date
 * @returns {Date}
 */
export const nextMonth = (date: Date) => {
  const isPassYear = date.getMonth() === 11;
  const newMonth = isPassYear ? 0 : date.getMonth() + 1;
  const year = isPassYear ? date.getFullYear() + 1 : date.getFullYear();
  return new Date(year, newMonth, 1);
};

/**
 * prev Month
 * @since version V1.0.0
 * @param {Date} date
 * @returns {Date}
 */
export const prevMonth = (date: Date) => {
  const passYear = date.getMonth() === 0;
  const newMonth = passYear ? 11 : date.getMonth() - 1;
  const year = passYear ? date.getFullYear() - 1 : date.getFullYear();
  return new Date(year, newMonth, 1);
};

/**
 * dateIndexOf
 * @since version V1.0.0
 * @param {Date} date
 * @param {Date[]} arrDates
 * @returns {Number}
 */
export function dateIndexOf(date: Date, arrDates: DateValue[]): number {
  const curDate = new Date(date);
  const curTime = curDate.setHours(0, 0, 0, 0);
  const selectedDates = arrDates.map((d: Date) => {
    const dt = new Date(d);
    const dtime = dt.setHours(0, 0, 0, 0);
    return dtime;
  });

  // console.log('curDate', selectedDates, curDate, selectedDates.indexOf(curTime));

  return selectedDates.indexOf(curTime);
}

/**
 * Empty value
 * @since version V1.0.0
 * @param mixedVar
 * @returns {Boolean}
 * @example
 * // example 1: empty(null)
 * // returns 1: true
 * // example 2: empty(undefined)
 * // returns 2: true
 * // example 3: empty([])
 * // returns 3: true
 * // example 4: empty({})
 * // returns 4: true
 * // example 5: empty({'fn' : function () {} })
 * // returns 5: false
 */
export function empty(mixedVar: any): mixedVar is undefined | null | false | '' | {} {
  let undef: undefined;
  const emptyValues = [undef, null, false, ''];
  for (let i = 0, len = emptyValues.length; i < len; i++) {
    if (mixedVar === emptyValues[i]) {
      return true;
    }
  }

  if (typeof mixedVar === 'object') {
    return Object.keys(mixedVar).length < 1;
  }

  return false;
}

/**
 * To Dashed from Camel Case
 * @since version V1.0.0
 * @param {String} strCamelCase
 * @returns {string}
 * @example
 * toDash('strCamelCase') === 'str-camel-case'
 */
export function toDash(strCamelCase: string): string {
  return String(strCamelCase)
    .split(/(?=[A-Z])/)
    .join('-')
    .toLowerCase();
}

export function setStyles(el: HTMLElement | SVGElement, styles: Styles) {
  Object.entries(styles).forEach(([prop, val = null]) => {
    if (empty(val)) {
      el.style.removeProperty(prop);
      return;
    }
    const [value, pri = ''] = String(val).split('!');

    el.style.setProperty(toDash(prop), value, pri);
  });
}

/**
 * 首字母大写
 * @param {String} str 目标字符串
 * @returns {String}
 */
export function firstUpperCase(str: string): string {
  if (!str) return str;
  return str[0].toUpperCase().concat(str.substring(1, str.length));
}

interface DateObj {
  year: number;
  month: number;
}

/**
 * 返回指定年、月的第一天日期
 * @param {Object} { year, month }
 * @returns {Date}
 */
function getFirstDayOfMonth({ year, month }: DateObj): Date {
  return new Date(year, month, 1);
}

/**
 * 返回指定年、月的天数
 * @param {Object} { year, month }
 * @returns {Number}
 */
function getDaysInMonth({ year, month }: DateObj): number {
  return new Date(year, month + 1, 0).getDate();
}

/**
 * 返回指定年、月的最后一天日期
 * @param {Object} { year, month }
 * @returns {Date}
 */
function getLastDayOfMonth({ year, month }: DateObj): Date {
  return new Date(year, month, getDaysInMonth({ year, month }));
}

function isSameYear(date1: Date, date2: Date) {
  return date1.getFullYear() === date2.getFullYear();
}

function isSameMonth(date1: Date, date2: Date) {
  return isSameYear(date1, date2) && date1.getMonth() === date2.getMonth();
}

function isSameDate(date1: Date, date2: Date) {
  return isSameMonth(date1, date2) && date1.getDate() === date2.getDate();
}

/**
 * 是否是某法范围内的日期，精确到日
 * @param {Date} value 目标日期
 * @param {Object} { start, end } 范围
 * @returns {Boolean}
 */
function isBetween(
  value: { getFullYear: () => number; getMonth: () => number; getDate: () => number },
  { start, end }: { start: any; end: any },
): boolean {
  const date = new Date(value.getFullYear(), value.getMonth(), value.getDate());

  const startTime = new Date(start.getFullYear(), start.getMonth(), start.getDate());
  const endTime = new Date(end.getFullYear(), end.getMonth(), end.getDate());

  return startTime <= date && endTime >= date;
}

/**
 * 比较两个日期对象的时间戳
 * @param {Date} date1 日期1
 * @param {Date} date2 日期2
 * @returns {Number} 返回 date1.getTime() - date2.getTime() 的差值
 */
function compareAsc(date1: { getTime: () => any }, date2: Date): number {
  const d1 = date1.getTime();
  const d2 = date2.getTime();

  if (d1 < d2) return -1;
  if (d1 > d2) return 1;
  return 0;
}

/**
 * 比较两个 Date 是否是同一天 或则 同一月 或则 同一年
 * @param {Date} date1 比较的日期
 * @param {Date} date2 比较的日期
 * @param {String} type 比较类型，默认比较到『日』 date|month|year
 * @returns {Boolean}
 */
export function isSame(date1: Date, date2: Date, type = 'date'): boolean {
  const func = {
    isSameYear,
    isSameMonth,
    isSameDate,
  };
  return func[`isSame${firstUpperCase(type)}`](date1, date2);
}

export function outOfRanges(d: Date, min: any, max: any) {
  return (min && compareAsc(d, min) === -1) || (max && compareAsc(d, max) === 1);
}

/**
 * @returns {Date} 当天零点的日期对象
 */
export function getToday(): Date {
  const now = new Date();
  return new Date(now.getFullYear(), now.getMonth(), now.getDate(), 0, 0, 0);
}

/**
 * 返回日期对象的年、月、日、小时、分钟、秒、12小时制标识
 * @param {Date} date
 * @returns {Object}
 */
export function getDateObj(date: Date) {
  if (!(date instanceof Date)) {
    throw new Error('date must be Date object');
  }
  return {
    year: date.getFullYear(),
    month: date.getMonth(),
    date: date.getDate(),
    hours: date.getHours(),
    minutes: date.getMinutes(),
    seconds: date.getSeconds(),
    meridiem: date.getHours() > 11 ? 'PM' : 'AM',
  };
}

/**
 * 设置日期对象的时间部分
 * @param {Date} d 日期
 * @param {Number} hour 小时
 * @param {Number} min 分钟
 * @param {Number} sec 秒
 * @returns {Date} 一个新的date
 */
export function setDateTime(d: Date, hour: number, min: number, sec: number): Date {
  // eslint-disable-next-line
  const { year, month, date } = getDateObj(d);
  return new Date(year, month, date, hour, min, sec, 0);
}

/**
 * 增加月份
 * @param {Date} date 起始日期
 * @param {Number} num 月份数
 * @returns {Date}
 */
export function subtractMonth(date: Date, num: any): Date {
  const day = date.getDate();
  const newDate = new Date(date);

  let monthNum = num;
  // eslint-disable-next-line no-plusplus
  while (monthNum--) {
    newDate.setDate(0);
  }
  newDate.setDate(day);
  return newDate;
}

/**
 * 减月份
 * @param {Date} date 起始日期
 * @param {Number} num 月份数
 * @returns {Date}
 */
export function addMonth(date: Date, num: number): Date {
  let monthNum = num;
  if (num < 0) monthNum = 0;
  const newDate = new Date(date);
  newDate.setMonth(date.getMonth() + monthNum);
  return newDate;
}

export interface OptionsType {
  firstDayOfWeek: number;
  disableDate: Function;
  minDate: Date;
  maxDate: Date;
  monthText?: string;
}

export function getWeeks(
  { year, month }: { year: number; month: number },
  { firstDayOfWeek, disableDate = () => false, minDate, maxDate }: OptionsType,
) {
  const prependDay = getFirstDayOfMonth({ year, month });

  const appendDay = getLastDayOfMonth({ year, month });

  const maxDays = getDaysInMonth({ year, month });

  const daysArr = [];

  const today = getToday();

  for (let i = 1; i <= maxDays; i++) {
    const currentDay = new Date(year, month, i);

    daysArr.push({
      text: i,
      active: false,
      value: currentDay,
      disabled: disableDate(currentDay) || outOfRanges(currentDay, minDate, maxDate),
      now: isSame(today, currentDay),
      firstDayOfMonth: i === 1,
      lastDayOfMonth: i === maxDays,
      type: 'current-month',
    });
  }

  if (prependDay.getDay() !== firstDayOfWeek - 1) {
    prependDay.setDate(0); // 上一月

    const FLAG = true;
    while (FLAG) {
      daysArr.unshift({
        text: prependDay.getDate().toString(),
        active: false,
        value: new Date(prependDay),
        disabled: disableDate(prependDay) || outOfRanges(prependDay, minDate, maxDate),
        additional: true, // 非当前月
        type: 'prev-month',
      });
      prependDay.setDate(prependDay.getDate() - 1);
      if (prependDay.getDay() === Math.abs(firstDayOfWeek + 6) % 7) break;
    }
  }

  const LEN = 42; // 显示6周

  while (daysArr.length < LEN) {
    appendDay.setDate(appendDay.getDate() + 1);
    daysArr.push({
      text: appendDay.getDate(),
      active: false,
      value: new Date(appendDay),
      disabled: disableDate(appendDay) || outOfRanges(appendDay, minDate, maxDate),
      additional: true, // 非当前月
      type: 'next-month',
    });
  }

  return chunk(daysArr, 7);
}

export function getYears(year: number, { disableDate = () => false, minDate, maxDate }: OptionsType) {
  const startYear = parseInt((year / 10).toString(), 10) * 10;
  const endYear = startYear + 9;

  const yearArr = [];

  const today = getToday();

  for (let i = startYear; i <= endYear; i++) {
    const date = new Date(i, 1);

    let disabledMonth = 0;
    let outOfRangeMonth = 0;

    for (let j = 0; j < 12; j++) {
      const d = new Date(i, j);
      if (disableDate(d)) disabledMonth += 1;
      if (outOfRanges(d, minDate, maxDate)) outOfRangeMonth += 1;
    }

    yearArr.push({
      value: date,
      now: isSame(date, today, 'year'),
      disabled: disabledMonth === 12 || outOfRangeMonth === 12,
      active: false,
      text: `${date.getFullYear()}`,
    });
  }

  return chunk(yearArr, 4);
}

export function getMonths(year: number, { disableDate = () => false, minDate, maxDate, monthText = '' }: OptionsType) {
  const monthArr = [];

  const today = getToday();

  for (let i = 0; i <= 11; i++) {
    const date = new Date(year, i);

    let disabledDay = 0;
    let outOfRangeDay = 0;

    const daysInMonth = getDaysInMonth({ year, month: i });

    for (let j = 1; j <= daysInMonth; j++) {
      const d = new Date(year, i, j);
      if (disableDate(d)) disabledDay += 1;
      if (outOfRanges(d, minDate, maxDate)) outOfRangeDay += 1;
    }

    monthArr.push({
      value: date,
      now: isSame(date, today, 'month'),
      disabled: disabledDay === daysInMonth || outOfRangeDay === daysInMonth,
      active: false,
      text: `${date.getMonth() + 1} ${monthText}`,
    });
  }

  return chunk(monthArr, 4);
}

interface DateTime {
  active: boolean;
  highlight: boolean;
  startOfRange: boolean;
  endOfRange: boolean;
  value: Date;
}

export function flagActive(data: any[], { ...args }: any) {
  const { start, end, type = 'date' } = args;

  if (!end) {
    return data.map((row: any[]) =>
      row.map((item: DateTime) => {
        Object.assign(item, { active: isSame(item.value, start, type) });
        return item;
      }),
    );
  }

  return data.map((row: any[]) =>
    row.map((item: DateTime) => {
      const date = item.value;
      const isStart = isSame(start, date, type);
      const isEnd = isSame(end, date, type);
      Object.assign(item, {
        active: isStart || isEnd,
        highlight: isBetween(date, { start, end }),
        startOfRange: isStart,
        endOfRange: isEnd,
      });
      return item;
    }),
  );
}
