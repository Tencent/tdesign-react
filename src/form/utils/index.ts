import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
