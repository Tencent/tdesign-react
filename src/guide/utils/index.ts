import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
