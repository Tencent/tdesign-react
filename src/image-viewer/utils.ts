export const downloadFile = function (imgSrc) {
  const image = new Image();
  const name = imgSrc?.split?.('/').pop() || Math.random().toString(32).slice(2);

  image.setAttribute('crossOrigin', 'anonymous');

  image.onload = () => {
    const canvas = document.createElement('canvas');
    canvas.width = image.width;
    canvas.height = image.height;

    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0, image.width, image.height);
    canvas.toBlob((blob) => {
      const url = URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.download = name;
      a.href = url;
      a.click();
      a.remove();
      URL.revokeObjectURL(url);
    });
  };
  image.src = imgSrc;
};
