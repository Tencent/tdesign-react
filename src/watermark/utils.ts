const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
