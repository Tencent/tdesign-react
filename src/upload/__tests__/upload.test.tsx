import { testExamples } from '@test/utils';
import MockDate from 'mockdate';

MockDate.set('2022-08-27');
// 测试组件代码 Example 快照
testExamples(__dirname);
