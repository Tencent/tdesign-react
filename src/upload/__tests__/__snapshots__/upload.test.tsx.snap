// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div>
    <div
      class="t-upload"
    >
      <input
        hidden=""
        type="file"
      />
      <div
        class="t-upload__single t-upload__single-file"
      >
        <div
          class="t-upload__trigger"
        >
          <button
            class="t-button t-button--theme-default t-button--variant-outline"
            type="button"
          >
            <svg
              class="t-icon t-icon-upload"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M3.74 6.68L7.5 2.9v8.59h1V2.91l3.76 3.77.71-.7-4.62-4.63a.5.5 0 00-.7 0L3.03 5.97l.7.7z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M2 11v2a1 1 0 001 1h10a1 1 0 001-1v-2h-1v2H3v-2H2z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
            <span
              class="t-button__text"
            >
              选择文件
            </span>
          </button>
        </div>
      </div>
      <small
        class="t-upload__tips t-size-s"
      >
        上传文件大小在 5M 以内
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`custom-drag.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-upload t-upload"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-outline"
      type="button"
    >
      <span
        class="t-button__text"
      >
        <svg
          class="t-icon t-icon-cloud-upload"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M4.73 6.26l-.71.09A2.32 2.32 0 002 8.67c0 1.2.89 2.18 2 2.31v1a3.31 3.31 0 01-3-3.31 3.32 3.32 0 012.9-3.32A4.2 4.2 0 018 2c2 0 3.69 1.43 4.1 3.35 1.63.2 2.9 1.6 2.9 3.32a3.31 3.31 0 01-3 3.32v-1c1.11-.14 2-1.11 2-2.32 0-1.22-.9-2.2-2.02-2.32l-.71-.09-.15-.7A3.2 3.2 0 008 3a3.2 3.2 0 00-3.12 2.56l-.15.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
          <path
            d="M6.14 10.72L7.5 9.39l.03 5.12 1.01-.01-.03-5.1 1.37 1.34.72-.7-2.26-2.2a.5.5 0 00-.7 0l-2.22 2.18.72.7z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        点击上传
      </span>
    </button>
    <br />
    <br />
    <div
      class="t-upload"
    >
      <input
        hidden=""
        type="file"
      />
      <div
        class="t-upload__dragger t-upload__dragger-center"
      >
        <div
          class="t-upload__trigger"
        >
          <button
            class="t-button t-button--theme-primary t-button--variant-base"
            type="button"
          >
            <span
              class="t-button__text"
            >
              自定义拖拽区域
            </span>
          </button>
          <br />
          <br />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`draggable.jsx 1`] = `
<DocumentFragment>
  <div>
    <div>
      是否自动上传：
      <button
        class="t-switch t-size-m"
        role="switch"
        type="button"
      >
        <span
          class="t-switch__handle"
        />
        <div
          class="t-switch__content"
        />
      </button>
      <br />
    </div>
    <div
      class="t-radio-group t-size-m t-radio-group__outline"
    >
      <label
        class="t-radio t-is-checked"
      >
        <input
          checked=""
          class="t-radio__former"
          type="radio"
          value="file"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          文件拖拽上传
        </span>
      </label>
      <label
        class="t-radio"
      >
        <input
          class="t-radio__former"
          type="radio"
          value="image"
        />
        <span
          class="t-radio__input"
        />
        <span
          class="t-radio__label"
        >
          图片拖拽上传
        </span>
      </label>
    </div>
    <br />
    <br />
    <div
      class="t-upload"
    >
      <input
        hidden=""
        type="file"
      />
      <div
        class="t-upload__dragger t-upload__dragger-center"
      >
        <div
          class="t-upload__trigger"
        >
          <div>
            <span
              class="t-upload--highlight"
            >
              点击上传
            </span>
            <span>
                /  拖拽到此区域
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`file-flow-list.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-upload__file-flow-demo"
  >
    <div
      class="t-upload"
    >
      <input
        hidden=""
        multiple=""
        type="file"
      />
      <div
        class="t-upload__flow t-upload__flow-file-flow"
      >
        <div
          class="t-upload__flow-op"
        >
          <div
            class="t-upload__trigger"
          >
            <button
              class="t-button t-button--theme-default t-button--variant-outline"
              type="button"
            >
              <svg
                class="t-icon t-icon-upload"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.74 6.68L7.5 2.9v8.59h1V2.91l3.76 3.77.71-.7-4.62-4.63a.5.5 0 00-.7 0L3.03 5.97l.7.7z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M2 11v2a1 1 0 001 1h10a1 1 0 001-1v-2h-1v2H3v-2H2z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
              <span
                class="t-button__text"
              >
                选择文件
              </span>
            </button>
          </div>
          <small
            class="t-upload__small t-upload__flow-placeholder"
          >
            支持批量上传文件，文件格式不限，最多只能上传 10 份文件
          </small>
        </div>
        <table
          class="t-upload__flow-table"
        >
          <thead>
            <tr>
              <th>
                文件名
              </th>
              <th>
                文件大小
              </th>
              <th>
                状态
              </th>
              <th>
                操作
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                colspan="4"
              >
                <div
                  class="t-upload__flow-empty"
                >
                  点击上方“选择文件”或将文件拖拽到此区域
                </div>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="t-upload__flow-bottom"
        >
          <button
            class="t-button t-button--theme-default t-button--variant-base"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消上传
            </span>
          </button>
          <button
            class="t-button t-button--theme-primary t-button--variant-base t-is-disabled"
            disabled=""
            type="button"
          >
            <span
              class="t-button__text"
            >
              点击上传
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`image.jsx 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="t-upload"
      >
        <input
          accept="image/*"
          hidden=""
          type="file"
        />
        <ul
          class="t-upload__card"
        >
          <li
            class="t-upload__card-item t-is--background"
          >
            <div
              class="t-upload__card-container t-upload__card-box"
            >
              <svg
                class="t-icon t-icon-add"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
              <p
                class="t-size-s"
              >
                点击上传图片
              </p>
            </div>
          </li>
        </ul>
        <small
          class="t-upload__tips t-size-s"
        >
          请选择单张图片文件上传
        </small>
      </div>
    </div>
    <br />
    <br />
    <div>
      <div
        class="t-upload"
      >
        <input
          accept="image/*"
          hidden=""
          type="file"
        />
        <ul
          class="t-upload__card"
        >
          <li
            class="t-upload__card-item t-is--background"
          >
            <div
              class="t-upload__card-content t-upload__card-box"
            >
              <img
                class="t-upload__card-image"
                src="http://0729iwiki-75822.gzc.vod.tencent-cloud.com/u=1595072465,3644073269&fm=193&f=GIF.jpeg"
              />
              <div
                class="t-upload__card-mask"
              >
                <span
                  class="t-upload__card-mask-item"
                >
                  <svg
                    class="t-icon t-icon-browse"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M10.88 8a2.88 2.88 0 11-5.76 0 2.88 2.88 0 015.76 0zm-1 0a1.88 1.88 0 10-3.76 0 1.88 1.88 0 003.76 0z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                    <path
                      d="M1.12 8.23A7.72 7.72 0 008 12.5c2.9 0 5.54-1.63 6.88-4.27L15 8l-.12-.23A7.73 7.73 0 008 3.5a7.74 7.74 0 00-6.88 4.27L1 8l.12.23zM8 11.5A6.73 6.73 0 012.11 8 6.73 6.73 0 0113.9 8 6.74 6.74 0 018 11.5z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                  </svg>
                </span>
                <span
                  class="t-upload__card-mask-item-divider"
                />
                <span
                  class="t-upload__card-mask-item"
                >
                  <svg
                    class="t-icon t-icon-delete"
                    fill="none"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M6 12V6h1v6H6zM9 6v6h1V6H9z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                    <path
                      d="M10.5 3H14v1h-1v10a1 1 0 01-1 1H4a1 1 0 01-1-1V4H2V3h3.5V1.8c0-.44.36-.8.8-.8h3.4c.44 0 .8.36.8.8V3zm-4 0h3V2h-3v1zM4 4v10h8V4H4z"
                      fill="currentColor"
                      fill-opacity="0.9"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </li>
        </ul>
        <small
          class="t-upload__tips t-size-s"
        >
          默认已上传文件
        </small>
      </div>
    </div>
    <br />
    <br />
    <div>
      <div
        class="t-upload"
      >
        <input
          accept="image/*"
          hidden=""
          multiple=""
          type="file"
        />
        <ul
          class="t-upload__card"
        >
          <li
            class="t-upload__card-item t-is--background"
          >
            <div
              class="t-upload__card-container t-upload__card-box"
            >
              <svg
                class="t-icon t-icon-add"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
              <p
                class="t-size-s"
              >
                点击上传图片
              </p>
            </div>
          </li>
        </ul>
        <small
          class="t-upload__tips t-size-s"
        >
          允许选择多张图片文件上传，最多只能上传 3 张图片
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`img-flow-list.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-upload__img-flow-demo"
  >
    <div
      class="t-upload"
    >
      <input
        accept="image/*"
        hidden=""
        multiple=""
        type="file"
      />
      <div
        class="t-upload__flow t-upload__flow-image-flow"
      >
        <div
          class="t-upload__flow-op"
        >
          <div
            class="t-upload__trigger"
          >
            <button
              class="t-button t-button--theme-default t-button--variant-outline"
              type="button"
            >
              <svg
                class="t-icon t-icon-upload"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.74 6.68L7.5 2.9v8.59h1V2.91l3.76 3.77.71-.7-4.62-4.63a.5.5 0 00-.7 0L3.03 5.97l.7.7z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M2 11v2a1 1 0 001 1h10a1 1 0 001-1v-2h-1v2H3v-2H2z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
              <span
                class="t-button__text"
              >
                选择文件
              </span>
            </button>
          </div>
          <small
            class="t-upload__small t-upload__flow-placeholder"
          >
            支持批量上传图片文件
          </small>
        </div>
        <div
          class="t-upload__flow-card-area"
        >
          <div
            class="t-upload__flow-empty"
          >
            点击上方“选择文件”或将文件拖拽到此区域
          </div>
        </div>
        <div
          class="t-upload__flow-bottom"
        >
          <button
            class="t-button t-button--theme-default t-button--variant-base"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消上传
            </span>
          </button>
          <button
            class="t-button t-button--theme-primary t-button--variant-base t-is-disabled"
            disabled=""
            type="button"
          >
            <span
              class="t-button__text"
            >
              点击上传
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`request-method.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column-large"
  >
    <div>
      <div
        class="t-radio-group t-size-m t-radio-group--filled"
      >
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            checked=""
            class="t-radio-button__former"
            type="radio"
            value="requestSuccessMethod"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            上传成功示例
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            type="radio"
            value="requestFailMethod"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            上传失败示例
          </span>
        </label>
        <div
          class="t-radio-group__bg-block"
          style="width: 0px; left: 0px;"
        />
      </div>
    </div>
    <div
      class="t-upload"
    >
      <input
        hidden=""
        type="file"
      />
      <div
        class="t-upload__single t-upload__single-file"
      >
        <div
          class="t-upload__trigger"
        >
          <button
            class="t-button t-button--theme-default t-button--variant-outline"
            type="button"
          >
            <svg
              class="t-icon t-icon-upload"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M3.74 6.68L7.5 2.9v8.59h1V2.91l3.76 3.77.71-.7-4.62-4.63a.5.5 0 00-.7 0L3.03 5.97l.7.7z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M2 11v2a1 1 0 001 1h10a1 1 0 001-1v-2h-1v2H3v-2H2z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
            <span
              class="t-button__text"
            >
              选择文件
            </span>
          </button>
        </div>
      </div>
      <small
        class="t-upload__tips t-size-s"
      >
        自定义上传方法需要返回成功或失败信息
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`single-custom.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-upload"
  >
    <div>
      <div
        class="t-upload"
      >
        <input
          hidden=""
          multiple=""
          type="file"
        />
        <div
          class="t-upload__trigger"
        >
          <button
            class="t-button t-button--theme-primary t-button--variant-base"
            type="button"
          >
            <span
              class="t-button__text"
            >
              自定义上传
            </span>
          </button>
        </div>
        <small
          class="t-upload__tips t-size-s"
        >
          上传文件大小在 1M 以内
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
